cask/src/cask/internal/VirtualThreadSupport.scala

```diff
@@ -0,0 +1,31 @@
+package cask.internal
+
+import java.lang.invoke.{MethodHandles, MethodType}
+import java.util.concurrent.ThreadFactory
+
+private[cask] object VirtualThreadSupport {
```

Comment:
This logic probably doesn't benefit from being in a separate file, and can be folded into a helper method on `VirtualThreadBlockingHandler`